### PR TITLE
Fix OTel Profiling proto implementation

### PR DESCRIPTION
**Pull request by florianl**
The OTel Profiling protocol implementation conflicts with the specification.

https://github.com/open-telemetry/opentelemetry-proto/blob/ff457cecf46cf219602e587d86d66f3b8cb3efe6/opentelemetry/proto/profiles/v1experimental/pprofextended.proto#L239-L245

This was detected and reported by https://github.com/elastic/otel-profiling-agent/issues/63. It does not resolve https://github.com/elastic/otel-profiling-agent/issues/63 as the handling of timestamps still need some attention with the OTel Profiling SIG. 

FYI: @brancz

### CODE REVIEW

**Review by rockdaboot — APPROVED** [review]

File: reporter/otlp_reporter.go [hunk 1]
```diff
@@ -697,10 +696,18 @@ func getSampleAttributes(profile *profiles.Profile, i traceFramesCounts) []uint6
...
+	}
+	if i.containerName != "" {
+		addAttr(semconv.ContainerNameKey, i.containerName)
+	}
+	if i.comm != "" {
+		addAttr(semconv.ThreadNameKey, i.comm)
+	}
+	if i.apmServiceName != "" {
+		addAttr(semconv.ServiceNameKey, i.apmServiceName)
+	}
```
Comment (athre0z) [inline reporter/otlp_reporter.go:710]: These checks are duplicated: `addAttr` already has an `if` for that.

Comment (rockdaboot) [reply reporter/otlp_reporter.go:710]: Maybe we should rename the function to `addAttrIfSet` or something!?

Comment (florianl) [reply reporter/otlp_reporter.go]: removed that commit.

**Comment by brancz** [comment]
lgtm

**Review by athre0z — APPROVED** [review]
LGTM!

**Pull request merged**